src/olympia/amo/models.py

```diff
@@ -522,11 +522,15 @@ def image_size(self):
     def delete_preview_files(cls, sender, instance, **kw):
         """On delete of the Preview object from the database, unlink the image
         and thumb on the file system """
-        for filename in [instance.image_path, instance.thumbnail_path]:
+        image_paths = [
+            instance.image_path, instance.thumbnail_path,
+            instance.original_path]
+        for filename in image_paths:
             try:
                 log.info('Removing filename: %s for preview: %s'
                          % (filename, instance.pk))
-                storage.delete(filename)
+                if storage.exists(filename):
+                    storage.delete(filename)
```

Comment:
Django actually implies that check already and does `os.path.exists` itself in `storage.delete`

Reply:
https://github.com/mozilla/addons-server/pull/8553/commits/d002da8dc8c33a60929d9ab9fec9643d928c28c2